server/accounts.go

```diff
@@ -1800,12 +2049,52 @@ func (a *Account) AddStreamImportWithClaim(account *Account, from, prefix string
 			prefix = prefix + string(btsep)
 		}
 	}
+	return a.AddMappedStreamImportWithClaim(account, from, prefix+from, imClaim)
+}
+
+// AddMappedStreamImport helper for AddMappedStreamImportWithClaim
+func (a *Account) AddMappedStreamImport(account *Account, from, to string) error {
+	return a.AddMappedStreamImportWithClaim(account, from, to, nil)
+}
+
+// AddMappedStreamImportWithClaim will add in the stream import from a specific account with optional token.
+func (a *Account) AddMappedStreamImportWithClaim(account *Account, from, to string, imClaim *jwt.Import) error {
+	if account == nil {
+		return ErrMissingAccount
+	}
+
+	// First check to see if the account has authorized export of the subject.
+	if !account.checkStreamImportAuthorized(a, from, imClaim) {
+		return ErrStreamImportAuthorization
+	}
+
+	if to == "" {
+		to = from
+	}
+	var (
+		usePub bool
+		tr     *transform
+		err    error
+	)
+	if subjectHasWildcard(from) {
+		if to == from {
+			usePub = true
+		} else {
+			// Create a transform
+			if tr, err = newTransform(from, transformTokenize(to)); err != nil {
+				a.mu.Unlock()
```

Comment:
I think remove this: I don't see the corresponding `a.mu.Lock()` above?

Reply:
Thx